content/post/deploy-and-manage-redis-sentinel-in-gke/index.md

```diff
@@ -0,0 +1,469 @@
+
+```bash
+$ kubectl create ns demo
+namespace/demo created
+```
+
+Here is the yaml of the Redis sentinel we are going to use:
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: RedisSentinel
+metadata:
+  name: sentinel
+  namespace: demo
+spec:
+  version: 7.0.5
+  replicas: 3
+  storageType: Durable
+  storage:
+    resources:
+      requests:
+        storage: 1Gi
+    storageClassName: "standard"
+    accessModes:
+    - ReadWriteOnce
+  terminationPolicy: WipeOut
+```
+
+Let's save this yaml configuration into `sentinel.yaml` 
+Then create the above Redis sentinel
+
+```bash
+$ kubectl create -f sentinel.yaml
+redissentinel.kubedb.com/sentinel created
+```
+In this yaml,
+* Here, we can see in the `spec.version` field specifies the version of Redis. Here, we are using Redis `version 7.0.5`. You can list the KubeDB supported versions of Redis by running `$ kubectl get redisversions` command.
+* Another field to notice is the `spec.storageType` field. This can be `Durable` or `Ephemeral` depending on the requirements of the database to be persistent or not.
+* Lastly, the `spec.terminationPolicy` field is *Wipeout* means that the database will be deleted without restrictions. It can also be "Halt", "Delete" and "DoNotTerminate". Learn More about these [HERE](https://kubedb.com/docs/latest/guides/redis/concepts/redis/#specterminationpolicy).
+
+Let's check the status of Redis sentinel,
+
+```bash
+$ kubectl get redissentinel -n demo
+NAME                                VERSION   STATUS   AGE
+redissentinel.kubedb.com/sentinel   7.0.5     Ready    7m9s
+```
+
+## Deploy Redis Cluster
+
+Now, we are going to deploy Redis cluster using KubeDB. Here is the yaml we are going to use,
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: Redis
+metadata:
+  name: redis
+  namespace: demo
+spec:
+  version: 7.0.5
+  replicas: 3
+  sentinelRef: 
+    name: sentinel
+    namespace: demo
+  mode: Sentinel
+  storageType: Durable
+  storage:
+    resources:
+      requests:
+        storage: 1Gi
+    storageClassName: "standard"
+    accessModes:
+    - ReadWriteOnce
+  terminationPolicy: WipeOut
+```
+
+Let's save this yaml configuration into `redis.yaml` 
+Then create the above Redis CRD
+
+```bash
+$ kubectl apply -f redis.yaml
+redis.kubedb.com/redis created
+```
+In this yaml,
+* Here, we can see in the `spec.version` field specifies the version of Redis. Here, we are using Redis `version 7.0.5`. You can list the KubeDB supported versions of Redis by running `$ kubectl get redisversions` command.
+* Another field to notice is the `spec.storageType` field. This can be `Durable` or `Ephemeral` depending on the requirements of the database to be persistent or not.
+* Lastly, the `spec.terminationPolicy` field is *Wipeout* means that the database will be deleted without restrictions. It can also be "Halt", "Delete" and "DoNotTerminate". Learn More about these [HERE](https://kubedb.com/docs/latest/guides/redis/concepts/redis/#specterminationpolicy).
+
```

Comment:
spec.SentinelRef.name and spec.SentinelRef.namespace specifies the sentinel instance which will monitor this redis database.